src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -404,13 +404,17 @@ otError MeshForwarder::UpdateMeshRoute(Message &aMessage)
     return error;
 }
 
-otError MeshForwarder::UpdateIp6RouteFtd(Ip6::Header &ip6Header, const Message &aMessage)
+otError MeshForwarder::UpdateIp6RouteFtd(Ip6::Header &ip6Header, Message &aMessage)
 {
     Mle::MleRouter &mle   = Get<Mle::MleRouter>();
     otError         error = OT_ERROR_NONE;
     Neighbor *      neighbor;
 
-    if (mle.IsRoutingLocator(ip6Header.GetDestination()))
+    if (aMessage.GetOffset() > 0)
+    {
+        mMeshDest = aMessage.GetMeshDest();
```

Comment:
This is used for direct msg tx only, correct?

Reply:
Yes, correct.